core/types/block.go

```diff
@@ -288,15 +288,17 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		b.header.Bloom = CreateBloom(receipts)
 	}
 
-	// We prevent setting the EmptyUncleHash only when we are on a migrated
-	// chain and the block is pre-gingerbread.
-	if len(uncles) == 0 && !(bType.IsMigratedChain() && !bType.IsGingerbread(header.Number)) {
-		b.header.UncleHash = EmptyUncleHash
-	} else {
-		b.header.UncleHash = CalcUncleHash(uncles)
-		b.uncles = make([]*Header, len(uncles))
-		for i := range uncles {
-			b.uncles[i] = CopyHeader(uncles[i])
+	// We prevent setting the unclehash when we are on a migrated chain and the
+	// block is pre-gingerbread.
+	if !(bType.IsMigratedChain() && !bType.IsGingerbread(header.Number)) {
+		if len(uncles) == 0 {
+			b.header.UncleHash = EmptyUncleHash
+		} else {
+			b.header.UncleHash = CalcUncleHash(uncles)
```

Comment:
I guess uncles must be empty if we **are** pre-gingerbread and on a migrated chain, so we don't have to copy uncles or set unclehash for that case?

Reply:
Yep, that is the logic, is it clear from the comment?

Reply:
Yes it's clear, just wanted to double check